### PR TITLE
Add gatsby-plugin-gatsby-cloud 1.0

### PR DESCRIPTION
**Pull request by dmccraw**
## Description

Add version 1.0 of `gatsby-plugin-gatsby-cloud`.

### CODE REVIEW

File: packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    header.replace(LINK_REGEX, (__, prefix, file, suffix) => {
+      const hashed = manifest[file]
+      if (hashed) {
+        return `${prefix}${pathPrefix}${hashed}${suffix}`
+      } else if (existsSync(publicFolder(file))) {
+        return `${prefix}${pathPrefix}${file}${suffix}`
+      } else {
+        throw new Error(
+          `Could not find the file specified in the Link header \`${header}\`.` +
+            `The gatsby-plugin-gastby-cloud is looking for a matching file (with or without a ` +
```
Comment (ascorbic) [inline packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js:186]:
```suggestion
            `The gatsby-plugin-gatsby-cloud is looking for a matching file (with or without a ` +
```

File: packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    return headers
+  }
+
+  const chunks = Array.from(pluginData.pages.values()).map(
+    page => page.componentChunkName
+  )
+
+  chunks.push(`pages-manifest`, `app`)
+
+  const files = [].concat(...chunks.map(chunk => pluginData.manifest[chunk]))
```
Comment (ascorbic) [inline packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js:314]: Why is this needed?

File: packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+      )
+      return `${text}${path}\n${headersString}`
+    },
+    ``
+  )
+}
+
+// program methods
+
+const validateUserOptions = (pluginOptions, reporter) => headers => {
```
Comment (ascorbic) [inline packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js:211]: Rather than validating this here, could this be done in [plugin options schema](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/configuring-usage-with-plugin-options/#how-to-validate-plugin-options)

**Review by ascorbic — COMMENTED** [review]
This is great. Thanks for the comprehensive tests. My main request would be that you use a plugin options schema for validating the options. There is also a lot of lodash in there where native JS would be fine. Not an issue in itself, but it does make it a bit harder to follow.

File: packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js (hunk 3 above)
Comment (sidharthachatterjee) [reply packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js:211]: This is a good idea

File: packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js (hunk 2 above)
Comment (sidharthachatterjee) [reply packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js:314]: We copied this over for the Netlify plugin. We'll review and iterate on this after release.

File: packages/gatsby-plugin-gatsby-cloud/.babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": [["babel-preset-gatsby-package", { "browser": true }]]
```
Comment (pieh) [inline packages/gatsby-plugin-gatsby-cloud/.babelrc:2]: NIT: but that's not browser plugin :P
```suggestion
  "presets": [["babel-preset-gatsby-package"]]
```

File: packages/gatsby-plugin-gatsby-cloud/package.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    "kebab-hash": "^0.1.2",
+    "lodash": "^4.17.20",
+    "webpack-assets-manifest": "^3.1.1"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.12.1",
+    "@babel/core": "^7.12.3",
+    "babel-preset-gatsby-package": "^1.0.0-next.0",
+    "cross-env": "^7.0.3",
+    "jest": "^26.0.1"
```
Comment (pieh) [inline packages/gatsby-plugin-gatsby-cloud/package.json:21]: this probably should be removed, our other package don't add `jest` directly and rely on `jest` provided by root of monorepo

Comment (sidharthachatterjee) [reply packages/gatsby-plugin-gatsby-cloud/package.json:21]: Removed

**Review by sidharthachatterjee — DISMISSED** [review]
Let's get this in. Cheers!

**Review by sidharthachatterjee — DISMISSED** [review]
🚢

**Review by sidharthachatterjee — APPROVED** [review]
This should be good to ship!
